src/afl-fuzz-run.c

```diff
@@ -921,6 +922,53 @@ common_fuzz_stuff(afl_state_t *afl, u8 *out_buf, u32 len) {
 
   fault = fuzz_run_target(afl, &afl->fsrv, afl->fsrv.exec_tmout);
 
+  int write_event_log = 1;
+  if (write_event_log) {
+    struct fuzz_event_buffer *buffer = &afl->event_buffer;
+    u64                       buf_size = buffer->size;
+    u64                       buf_max_len = 16384;
+    if (buf_max_len <= buf_size) {
+      // TODO(wuestholz): Trigger this before AFL terminates.
+      FILE *f = buffer->log_file;
+      if (!f) {
+        f = fopen(alloc_printf("%s/fuzz_event_log.csv", afl->out_dir), "w");
+        if (!f) { PFATAL("Unable to create 'fuzz_event_log.csv'"); }
+        buffer->log_file = f;
+      }
+
+      for (u64 i = 0; i < buf_max_len; i++) {
+        struct fuzz_event event = buffer->events[i];
+        // We reduce the size of the checksums to save space.
+        u32 oc32 = event.original_cksum - (event.original_cksum >> ((u64)32));
+        u32 fc32 = event.fuzzed_cksum - (event.fuzzed_cksum >> ((u64)32));
```

Comment:
Why not simply truncate by casting into something smaller?
Better, why not define original_cksum or fuzzed_cksum to be smaller (u16 or u32)?

Reply:
I think I was trying to use all bits of the original hash, but casting should be fine for our purpose. I added the size reduction mostly as a way to optimize the output, but I guess we could already perform the reduction earlier when storing the checksums in memory.